incubator/types-core/src/main/java/io/aklivity/zilla/runtime/types/core/StringValidatorHandler.java

```diff
@@ -77,31 +73,19 @@ public boolean validate(
                     }
                 }
                 index += charByteCount;
-
-                if (index < limit && (charByte0 & 0b11000000) != 0b10000000)
-                {
-                    incompleteCharCount = charByteCount - 1;
-                    incompleteChar = charByte0 & (0b111111 >> charByteCount);
-                    break;
-                }
             }
         }
 
+        pendingCharBytes = limit - index;
+
         if ((flags & FLAGS_FIN) != 0x00)
         {
-            valid = initialize && incompleteCharCount == 0 && index == limit;
-            initialize = false;
+            valid = pendingCharBytes == 0 && index == limit;
         }
         else
         {
-            valid = initialize && index == limit;
+            valid = index + pendingCharBytes == limit;
         }
```

Comment:
Suggest simplifying to a ternary operator.
```
value = (flags & FLAGS_FIN) == 0x00
    ? index + pendingCharBytes == limit
    : pendingCharBytes == 0 && index == limit;
```

Btw, why is `index + pendingCharBytes == limit` the correct expression for fragmented messages? The `index` and `length` represent a single fragment, so why does that relate to `pendingBytes`?